doc/LogitechSpotlightHID++.md

```diff
@@ -0,0 +1,137 @@
+
+For the sake of completeness, it should be noted that the official Logitech Spotlight software reprogram the click and double click events too by following HID++ commands:
+
+```
+// Send click event as HID++ message (0x07 - Feature Index for ReprogramControlsV4 Feature Code, 0xd8-click button, 0x33- hold event)
+{0x11, 0x01, 0x07, 0x3d, 0x00, 0xd8, 0x33, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00}
+// Send double click as HID++ message (0x07 - Feature Index for ReprogramControlsV4 Feature Code, 0xdf - double click)
+{0x11, 0x01, 0x07, 0x3d, 0x00, 0xdf, 0x03, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00}
+```
+
+However, Projecteur does not send these packages. Instead it grab the events from mouse event device associated with Spotlight. The approach taken by Projecteur is advantageous as it help in implementing Input Mapping feature that official Logitech sofware lacks. However, this approach also make porting Projecteur to different platform somewhat difficult.
+
+# Wireless Notification on Activation/Deactivation of Spotlight Device
+
+The Spotlight device send wireless notification if it get activated. Wireless notification will be short HID++ message if Spotlight device is connected through USB dongle. Otherwise, it will be a long HID++ message.
+
+For short HID++ wireless notification, the third byte will be 0x41. In such message, 6th bit in 5th byte show activation status of spotlight device. If the 6th bit is 0 then device got active otherwise the device got deactivated.
+
+Long HID++ wireless notification is only received for device activation. In such message, third byte will be Feature Index for Wireless Notification Feature Code.
+
+# Vibration support
+
+The spotlight device can vibrate if HID++ message {0x10, 0x01, (Feature Index for Presenter Control Feature Code), 0x1d, length, 0xe8, intensity} is sent to it. In the message, length can range between 0x00 to 0x0a.
+
+# Note
+
+All of HID++ commands listed above result in response from Spotlight device. Many a time these response are just the acknowledgement of the HID++ commands sent from PC. However, in some of response Spotlight device can send useful information. To check processing of these responses please check the `onHidppDataAvailable` method in `Spotlight` class in [spotlight.h](../src/spotlight.h).
+
+For more information about HID++ protocol in general please check files uploaded to [this folder](https://drive.google.com/drive/folders/0BxbRzx7vEV7eWmgwazJ3NUFfQ28?resourcekey=0-dQ-Lx1FORQl0KAdOHQaE1A) or checkout [logitech-hidpp module](https://github.com/torvalds/linux/blob/master/drivers/hid/hid-logitech-hidpp.c) code in linux kernel source.
```

Comment:
I don't like linking to some google drive link for legal reasons
1. I don't know what documents are on there 
2. Event if the documents that are on there are totally fine right now, I don't know what happens in the future and I can't monitor that drive share.